### PR TITLE
ZK Problems

### PR DESCRIPTION
**Pull request by colinmarc**
This downgrades the vendored zookeeper client to 3.4.8 (it was 
from HEAD), and fixes some assorted zookeeper issues.

The biggest change is to increase the timeout. 100ms or 1s is 
way too low of a timeout - I just observed a zookeeper fsync for
1.4s. That may be another issue, but it looks like storm sets 20s
by default, so I think we're in good company with 10s. There's not
really a downside to having a larger timeout, in any case.

This also tweaks some logic; it changes to the reconnect logic such
that it doesn't bother trying to reestablish the session, and it
ensures that we don't try to set watches while we're reconnecting.

### CODE REVIEW

File: zk_watcher.go [hunk 1]
```diff
@@ -112,9 +107,6 @@ func (w *zkWatcher) reconnect() error {
 		return err
 	}
 
-	log.Println("Successfully connected to zookeeper!")
```
Comment (forklady42) [inline zk_watcher.go]: keep this log? we've generally wanted more logging rather than less so far

**Comment by forklady42** [comment]
lgtm 👍

**Comment by colinmarc** [comment]
@forklady42 I added two more tweaks: 2b3b979 and 4def7b2. Want to take another look?

**Comment by forklady42** [comment]
Updates look 😎

**Pull request merged**